components/markdown_image.test.jsx

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import configureStore from 'redux-mock-store';
+import {shallow} from 'enzyme';
+
+import MarkdownImage from 'components/markdown_image.jsx';
+
+describe('components/MarkdownImage', () => {
+    const mockStore = configureStore();
+    const initialState = {
+        entities: {
+            general: {
+                config: {
+                    DefaultClientLocale: 'en',
+                },
+            },
+        },
+    };
+    const store = mockStore(initialState);
+    it('should match snapsnot for SizeAwareImage dimensions', () => {
+        const imageMetadata = {format: 'jpg', frame_count: 0, width: 100, height: 100};
+        const wrapper = shallow(
+            <MarkdownImage
+                imageMetadata={imageMetadata}
+                src='path/image'
+            />
+        );
+
+        const childrenWrapper = wrapper.dive({context: {store}}).props().children('safeSrc');
+
+        expect(childrenWrapper).toMatchSnapshot();
```

Comment:
Maybe instead of using a snapshot here, you could just check whether or not the dimensions are set? That's the only thing you care about testing here, so it'd be nice to be clear about that

```suggestion
        expect(childrenWrapper.prop('dimensions')).toBe(imageMetadata);
```

Reply:
Yes, that is what i wanted to achieve but i couldn't figure out a way to do this. I tried various versions of this with no luck.

Reply:
@hmhealey That brings to my first question in the PR.

Reply:
I had suggested `dive` since I thought it would call the inner function for you, but calling `wrapper.props().children` directly seems like it works even better since you don't need to initialize the store